openassessment/xblock/submission_mixin.py

```diff
@@ -203,13 +209,23 @@ def upload_url(self, data, suffix=''):
             A URL to be used to upload content associated with this submission.
 
         """
-        if "contentType" not in data:
-            return {'success': False, 'msg': self._(u"Must specify contentType.")}
+        if 'contentType' not in data or 'filename' not in data:
+            return {'success': False, 'msg': self._(u"There was an error uploading your file.")}
         content_type = data['contentType']
+        file_name = data['filename']
+        file_ext = file_name.split('.')[-1]
```

Comment:
os.path.splitext is useful for splitting file extensions.  It will give "" for a file name like "abc", this code will give "abc".